lib/src/db/db.dart

```diff
@@ -237,15 +240,24 @@ class Query<T extends Model> {
         offset: _offset,
         limit: _limit);
 
-    ds.Partition partition;
+    ds.Partition? partition;
     if (_partition != null) {
-      partition = ds.Partition(_partition.namespace);
+      partition = ds.Partition(_partition!.namespace);
     }
 
     return StreamFromPages<ds.Entity>((int pageSize) {
-      return _db.datastore
-          .query(query, transaction: _transaction, partition: partition);
-    }).stream.map<T>(_db.modelDB.fromDatastoreEntity);
+      if (_transaction != null) {
```

Comment:
Nit: why was this needed? I'd thought that you could have passed `transaction: _transaction, partition: partition` as before.

Reply:
We can, if we allow `transaction` to be nullable... but is that a better choice? maybe it's fine...